plugins/modules/packaging/os/pkgutil.py

```diff
@@ -96,117 +149,129 @@ def run_command(module, cmd, **kwargs):
     return module.run_command(cmd, **kwargs)
 
 
-def package_install(module, state, name, site, update_catalog):
-    cmd = ['pkgutil', '-iy']
+def package_install(module, state, pkgs, site, update_catalog, force):
+    cmd = ['pkgutil']
+    if module.check_mode:
+        cmd.append('-n')
+    cmd.append('-iy')
     if update_catalog:
-        cmd += ['-U']
+        cmd.append('-U')
```

Comment:
If both `-n` and `-U` are specified, does it modify the state somehow?

Reply:
No.  We can see from the `pkgutil` [source code ](https://sourceforge.net/p/pkgutil/code/HEAD/tree/trunk/pkgutil) that it exits just before fetching the catalog if `-n` is specified.  This isn't directly affected by `-U`.

Note however that the catalog may get updated anyway even in check mode if `state` is `latest`, because that may happen as a part of checking for the latest.  I think this is reasonable and expected.